app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassJavaActivity.java

```diff
@@ -0,0 +1,149 @@
+package com.getsensibill.spendmanager.demo.subclass;
+
+import android.os.Bundle;
+
+import com.getsensibill.spendmanager.demo.databinding.ActivitySubclassJavaBinding;
+import com.getsensibill.web.data.UiFinishReason;
+import com.getsensibill.web.data.configuration.NavigationIntent;
+import com.getsensibill.web.data.configuration.ProgrammaticTheme;
+import com.getsensibill.web.data.models.Brand;
+import com.getsensibill.web.ui.WebUiActivity;
+import com.getsensibill.web.ui.WebUiFragment;
+import com.getsensibill.web.ui.WebUiNetworkErrorFragment;
+
+import org.jetbrains.annotations.NotNull;
+
+import timber.log.Timber;
+
+/**
+ * Subclassing the WebUiActivity
+ * <p>
+ * When you subclass the WebUiActivity, you do not need to override anything for it to work.
+ * It will use all the defaults provided by the [WebUiActivity] and the [WebUiFragment].
+ *
+ * If you are using your own layout, then its important to override the [callSetContentView] and
+ * the [webContainerId] so that the [WebUiActivity] knows where to place the [WebUiFragment].
+ */
+public class SubclassJavaActivity extends WebUiActivity implements WebUiFragment.Listener {
+
+    private ActivitySubclassJavaBinding binding;
+
+    /**
+     * Not required override.
+     *
+     * Can be used to override the setContentView, so that it uses your provided
+     * activities view, instead of the default [WebUiActivity] view.
+     */
+    @Override
+    protected void callSetContentView() {
+        binding = ActivitySubclassJavaBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+    }
+
+    /**
+     * Not required override, unless you are overriding [callSetContentView]
+     *
+     * If we are overriding the setContentView, we also need to set the container id
+     * which will contain the [WebUiFragment].
+     */
+    @Override
+    protected int getWebContainerId() {
+        return binding.webContainer.getId();
```

Comment:
Should we do this or `R.id.webContainer`?  I guess i just never thought about it but my first thought would be to use R